src/Extensions/Compatability/Logger/Category.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using Microsoft.Omex.Extensions.Compatability.Validation;
+
+namespace Microsoft.Omex.Extensions.Compatability.Logger
+{
+	/// <summary>
+	/// Logging category
+	/// </summary>
+	public struct Category : IEquatable<Category>
+	{
+		/// <summary>
+		/// Category
+		/// </summary>
+		/// <param name="name">name</param>
+		/// <exception cref="ArgumentNullException"><paramref name="name"/> is null or white space</exception>
+		public Category(string name)
+		{
+			Name = Code.ExpectsNotNullOrWhiteSpaceArgument(name, nameof(name), null);
```

Comment:
>ExpectsNotNullOrWhiteSpaceArgument [](start = 15, length = 34)

Can `name` actually be `null` based on your other comment about using non-nullable types

Reply:
Thank you, It is a copy from all code, I'll update it to take null logic into account

Reply:
I think I may have created more confusion - sorry. `name` can still be empty or whitespace-only, so the check was adding something. We can probably live without the check, but it may have been adding value

---
In reply to: [362464826](https://github.com/microsoft/Omex/pull/153#discussion_r362464826) [](ancestors = 362464826)

Reply:
I've checked old code, and in case if the empty string it will replace it with null :) so I think it's better without it since this class added only for compatibility